src/runtime/pprof/pprof_test.go

```diff
@@ -1230,6 +1230,50 @@ func TestMutexProfile(t *testing.T) {
 	})
 }
 
+func TestMutexProfileChangeRate(t *testing.T) {
+	old := runtime.SetMutexProfileFraction(1)
+	defer runtime.SetMutexProfileFraction(old)
+
+	countcontentions := func() int64 {
+		var w bytes.Buffer
+		Lookup("mutex").WriteTo(&w, 0)
+		p, err := profile.Parse(&w)
+		if err != nil {
+			t.Fatalf("failed to parse profile: %v", err)
+		}
+		t.Logf("parsed proto: %s", p)
+		if err := p.CheckValid(); err != nil {
+			t.Fatalf("invalid profile: %v", err)
+		}
+
+		var count int64
+		for _, s := range p.Sample {
+			count += s.Value[0]
+		}
+		return count
+	}
+
+	original := countcontentions()
+	newrate := 2
+	for i := 0; i < newrate*50; i++ {
+		blockMutex(t)
+	}
+	before := countcontentions() - original
+	runtime.SetMutexProfileFraction(newrate)
+	for i := 0; i < 50; i++ {
+		blockMutex(t)
+	}
+	after := countcontentions() - original
+
+	// We shouldn't get more than twice as many contentions reported after
+	// than we got before. If we adjust the contentions by the sampling
+	// probability 1/newrate, then we expect at most 50*newrate additional
+	// contentions from the second set of blockMutex calls.
+	if after > 2*before {
```

Comment:
Let me make sure I understand this. `before` is expected to be exactly 100, and `after` is expected to be between `[100,200]` (inclusive) with an expected mean of `150`, right? Maybe that would be a more clear way to write this comment?

Reply:
Yeah, that's a better way to put it. I can update the comment to make that more clear.